### PR TITLE
LPS-72206 Add support for IANA links and not-embeddable relations

### PR DESCRIPTION
**Pull request by ahdezma**

### CODE REVIEW

**Comment by ahdezma** [comment]
ci:stop

**Comment by ahdezma** [comment]
ci:retest

**Comment by ahdezma** [comment]
ci:retest

File: vulcan-api/src/main/java/com/liferay/vulcan/representor/builder/RepresentorBuilder.java [hunk 1]
```diff
@@ -35,6 +35,12 @@
 		public FirstStep<T> addField(
 			String key, Function<T, Object> valueFunction);
 
+		public <S> FirstStep<T> addLink(
+			String key, Class<S> clazz,
+			Function<T, Optional<S>> objectFunction);
+
+		public FirstStep<T> addLink(String key, String link);
```
Comment (JorgeFerrer) [inline vulcan-api/src/main/java/com/liferay/vulcan/representor/builder/RepresentorBuilder.java]: The second parameter should be called url

File: vulcan-api/src/main/java/com/liferay/vulcan/representor/builder/RepresentorBuilder.java [hunk 2]
```diff
@@ -35,6 +35,12 @@
 		public FirstStep<T> addField(
 			String key, Function<T, Object> valueFunction);
 
+		public <S> FirstStep<T> addLink(
+			String key, Class<S> clazz,
```
Comment (JorgeFerrer) [inline vulcan-api/src/main/java/com/liferay/vulcan/representor/builder/RepresentorBuilder.java]: I think we should rename clazz to modelClass here and in the addEmbedded() method.

**Comment by ahdezma** [comment]
ci:retest

**Comment by ahdezma** [comment]
ci:retest

**Pull request closed without merging**